### PR TITLE
Fix clippy 1.67 warnings

### PR DESCRIPTION
**Pull request by mimame**
This is my first commit using Rust and to Helix, so I hope to be useful :smile:

### CODE REVIEW

**Comment by archseer** [comment]
1.67 uses features which aren't available on our MSRV. Also the formatting string lint should be disabled: https://twitter.com/adam_chal/status/1618673969590304775

**Comment by archseer** [comment]
https://github.com/rust-lang/rust-clippy/issues/10087

**Comment by mimame** [comment]
Hi @archseer, I've reduced the changes to only MSRV warnings, If I'm not wrong 
Thanks so much for your review!

File: helix-term/src/ui/menu.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ impl<T: Item> Menu<T> {
         Self {
             options,
             editor_data,
-            matcher: Box::new(Matcher::default()),
+            matcher: Box::<fuzzy_matcher::skim::SkimMatcherV2>::default(),
```
Comment (pascalkuthe) [inline helix-term/src/ui/menu.rs:80]: You don't need the turbo fish (type specification) here. `Bix::default()` should be enough and is much more readable

File: helix-term/src/ui/picker.rs [hunk 2]
```diff
@@ -431,7 +431,7 @@ impl<T: Item> Picker<T> {
         let mut picker = Self {
             options,
             editor_data,
-            matcher: Box::new(Matcher::default()),
+            matcher: Box::<fuzzy_matcher::skim::SkimMatcherV2>::default(),
```
Comment (pascalkuthe) [inline helix-term/src/ui/picker.rs:434]: Same here

File: helix-term/src/ui/menu.rs (hunk 1 above)
Comment (mimame) [reply helix-term/src/ui/menu.rs]: Thanks so much @pascalkuthe for your review! 
Fixed

File: helix-term/src/ui/picker.rs (hunk 2 above)
Comment (mimame) [reply helix-term/src/ui/picker.rs]: Thanks so much @pascalkuthe for your review! 
Reverted

File: helix-term/src/ui/menu.rs (hunk 1 above)
Comment (pascalkuthe) [reply helix-term/src/ui/menu.rs]: You don't need to revert this changes entirely.
You reverted to `Box::new(Matcher::default())` but I mean that you can just use `Box::default()` here. 
With my comment I just ment that you don't need the `::<fuzzy_matcher::skim::SkimMatcherV2>` part

Comment (mimame) [reply helix-term/src/ui/menu.rs]: Thanks for the clarification.
Fixed

**Review by pascalkuthe — APPROVED** [review]
LGTM

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**